Library/Homebrew/Gemfile

```
@@ -13,6 +13,8 @@ gem "rspec-retry", require: false
 gem "rspec-wait", require: false
 gem "rubocop"
 gem "simplecov", require: false
+gem "sorbet"
+gem "sorbet-runtime"
```

```suggestion
if ENV["HOMEBREW_SORBET"]
  gem "sorbet"
  gem "sorbet-runtime"
end
```

Might be nice to do this and leave the `Gemfile.lock` and others as-is until we have code that's going to be used by more users. Thoughts? Does that work as expected for you?

---

If we're going to go with using Sorbet - which we are, otherwise we wouldn't have a GSoC project for it - then I'd be inclined to make this as normal as possible as early as possible so we have less to unpick to make this mainstream later.

Incremental is good, but I thought that meant "only enabling it everywhere slowly, and on CI at the very end", etc.

That said, keeping it behind an envvar does restrict it to people who want it (that is: Vidushee, us) and might discourage enthusiastic contributors from seeing Sorbet in the Gemfile, running `srb tc` and doing the work themselves. But I don't know how many people would actually do that - the PRs do have labels on, after all.

---

> If we're going to go with using Sorbet - which we are, otherwise we wouldn't have a GSoC project for it - then I'd be inclined to make this as normal as possible as early as possible so we have less to unpick to make this mainstream later.

I agree but there's similar discussions in https://github.com/Homebrew/brew/pull/7678 and I think it's important we're being consistent. 

I think we want to keep the number of gems we're installing for contributors who want to e.g. just run `brew style` at a minimum until they are going to interact with that functionality.



> might discourage enthusiastic contributors from seeing Sorbet in the Gemfile, running `srb tc` and doing the work themselves.

I think we do want to discourage that to avoid conflicts/stepping on toes.

---

> I agree but there's similar discussions in https://github.com/Homebrew/brew/pull/7678 and I think it's important we're being consistent.

In that case, I'm in favour of consistency as long as all projects (that touch gems) do the same.

> I think we do want to discourage that to avoid conflicts/stepping on toes.

Definitely. OK, let's do it this way (provided it works).

---

Just checked and it works fine. I think it will be okay to keep the gemfile this way till we don't add method signatures. 👍